third-party/cmake/BuildLibtermkey.cmake

```diff
@@ -43,6 +43,7 @@ ExternalProject_Add(libtermkey
                               PREFIX=${DEPS_INSTALL_DIR}
                               PKG_CONFIG_PATH=${DEPS_LIB_DIR}/pkgconfig
                               CFLAGS=-fPIC
+                              CFLAGS+=-g
```

Comment:
But this is not debug-with-optimizations, it's just debug. We set `HAS_OG_FLAG` in `CMakeLists.txt`, can that be used to add `-Og -g` if supported?

Reply:
I'm confused. Are we talking about Debug or ReleaseWithDebInfo now?

Reply:
The short answer is: just change this line to `CFLAGS+=-Og CFLAGS+=-g`.

The long answer is: `-Og` is not supported by some compilers/versions, so `-Og` should not be included. `HAS_OG_FLAG` checks whether the compiler supports `-Og`.

> Are we talking about Debug or ReleaseWithDebInfo now? What am I supposed to do for which build type?

I don't understand the question, we aren't doing anything different for different build-types.  We're just trying to enable optimizations if possible.

Reply:
How about this?

```diff
diff --git i/third-party/CMakeLists.txt w/third-party/CMakeLists.txt
index ceebb4cf1..7404ca82c 100644
--- i/third-party/CMakeLists.txt
+++ w/third-party/CMakeLists.txt
@@ -4,0 +5,3 @@ project(NVIM_DEPS)
+# Used by check_c_compiler_flag()
+include(CheckCCompilerFlag)
+
@@ -13,0 +17,7 @@ endif()
+set(DEFAULT_C_FLAGS "-g -fPIC")
+
+check_c_compiler_flag(-Og HAS_OG_FLAG)
+if(HAS_OG_FLAG)
+  set(DEFAULT_C_FLAGS "-0g ${DEFAULT_C_FLAGS}")
+endif()
+
diff --git i/third-party/cmake/BuildLibtermkey.cmake w/third-party/cmake/BuildLibtermkey.cmake
index a99949e65..167692eee 100644
--- i/third-party/cmake/BuildLibtermkey.cmake
+++ w/third-party/cmake/BuildLibtermkey.cmake
@@ -45,2 +45 @@ ExternalProject_Add(libtermkey
-                              CFLAGS=-fPIC
-                              CFLAGS+=-g
+                              CFLAGS=${DEFAULT_C_FLAGS}
diff --git i/third-party/cmake/BuildLibvterm.cmake w/third-party/cmake/BuildLibvterm.cmake
index d6754f09b..76affb757 100644
--- i/third-party/cmake/BuildLibvterm.cmake
+++ w/third-party/cmake/BuildLibvterm.cmake
@@ -50,4 +50,3 @@ else()
-                                PREFIX=${DEPS_INSTALL_DIR}
-                                CFLAGS=-fPIC
-                                CFLAGS+=-g
-                                install)
+                                           PREFIX=${DEPS_INSTALL_DIR}
+                                           CFLAGS=${DEFAULT_C_FLAGS}
+                                           install)
diff --git i/third-party/cmake/BuildUnibilium.cmake w/third-party/cmake/BuildUnibilium.cmake
index 2b1084775..f56c1af8c 100644
--- i/third-party/cmake/BuildUnibilium.cmake
+++ w/third-party/cmake/BuildUnibilium.cmake
@@ -22,2 +22 @@ ExternalProject_Add(unibilium
-                            CFLAGS=-fPIC
-                            CFLAGS+=-g
+                            CFLAGS=${DEFAULT_C_FLAGS}
```

I just need to figure out a way how to quote arguments properly. Obviously the argument to `CFLAGS` here needs to be quoted. But.. easier said than done in cmake!

Reply:
@mhinz isn't there a way to pass `HAS_OG_FLAG` from the top-level CMakeLists.txt to the `third-party/CMakeLists.txt` sub-project?

> how to quote arguments properly

AFAIK CMake tries to magically quote things on its own, so it's better to omit quotes than try to do them yourself.

Reply:
Well, it's not a sub-project. They're two separate projects called by our Makefile. But [`load_cache()`](https://cmake.org/cmake/help/v3.0/command/load_cache.html) looks like it could be used instead.

The problem with the quoting above is that `CFLAGS=${DEFAULT_C_FLAGS}` leads to `make CFLAGS=-Og -g` which doesn't work. But `CFLAGS="${DEFAULT_C_FLAGS}`" leads to `make "CFLAGS=-Og -g`, which obviously doesn't work either. Cmake argument quoting is mayhem.

Reply:
Another thing that just came to my mind.. with `load_cache()` we would reference another project and might include things we don't want. Yes, `load_cache()` provides a way to exclude entries, but it seems like a rabbit hole to me.

Moreover, deps are built before core, so after a `make clean` we would have no `HAS_OG_FLAG` set yet, so we would need to run the code anyway.